File: StayFit/Models/CoachClient.cs
using StayFit.Enums;
using System.ComponentModel.DataAnnotations.Schema;

namespace StayFit.Models
{
    public class CoachClient
    {
        public long Id { get; set; }
        public CoachClientStatus Status { get; set; }
        public DateTime RequestedAt { get; set; }

        [ForeignKey("ClientProfile")]
        public long ClientProfileId { get; set; }
        public ClientProfile ClientProfile { get; set; } = null!;
        
        [ForeignKey("CoachProfile")]
        public long CoachProfileId { get; set; }
        public CoachProfile CoachProfile { get; set; } = null!;
    }
}

File: Frontend/stayfit/src/app/core/services/coach-client.ts
import { Injectable } from '@angular/core';
import { BaseApi } from './base-api';
import { CoachClientStatus } from '../models/enums';

export interface CoachClientDto {
  id: number;
  clientProfileId: number;
  userId: number;
  name: string;
  status: CoachClientStatus;
  requestedAt: string;
}

@Injectable({ providedIn: 'root' })
export class CoachClient extends BaseApi {
  requestCoach(coachId: number) {
    return this.post<void>(`/coach-client/request/${coachId}`, {});
  }

  getMyClients() {
    return this.get<CoachClientDto[]>('/coach-client/my-clients');
  }

  getMyCoaches() {
    return this.get<CoachClientDto[]>('/coach-client/my-coaches');
  }

  accept(id: number) {
    return this.put<void>(`/coach-client/${id}/accept`, {});
  }

  reject(id: number) {
    return this.put<void>(`/coach-client/${id}/reject`, {});
  }

  endRelation(id: number) {
    return this.delete<void>(`/coach-client/${id}`);
  }
}
